src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1810 @@
+            stride=1,
+            padding=(config.speech_decoder_postnet_kernel - 1) // 2,
+            bias=False,
+        )
+        self.batch_norm = nn.BatchNorm1d(out_conv_dim)
+
+        if layer_id < config.speech_decoder_postnet_layers - 1:
+            self.activation = nn.Tanh()
+        else:
+            self.activation = None
+
+        self.dropout = nn.Dropout(config.speech_decoder_postnet_dropout)
+
+    def forward(self, hidden_states):
+        hidden_states = self.conv(hidden_states)
+        hidden_states = self.batch_norm(hidden_states)
+        if self.activation is not None:
+            hidden_states = self.activation(hidden_states)
+        hidden_states = self.dropout(hidden_states)
+        return hidden_states
+
+
+class FastSpeech2ConformerSpeechDecoderPostnet(nn.Module):
+    def __init__(self, config):
+        super().__init__()
+        self.config = config
+        self.feat_out = nn.Linear(config.hidden_size, config.num_mel_bins * config.reduction_factor)
+        self.layers = nn.ModuleList(
+            [FastSpeech2ConformerBatchNormConvLayer(config, i) for i in range(config.speech_decoder_postnet_layers)]
+        )
+
+    def forward(self, hidden_states: torch.Tensor):
+        outputs_before_postnet = self.feat_out(hidden_states).view(hidden_states.size(0), -1, self.config.num_mel_bins)
+        outputs_after_postnet = self.postnet(outputs_before_postnet)
+        return outputs_before_postnet, outputs_after_postnet
+
+    def postnet(self, hidden_states: torch.Tensor):
+        layer_output = hidden_states.transpose(1, 2)
+        for layer in self.layers:
+            layer_output = layer(layer_output)
+        return hidden_states + layer_output.transpose(1, 2)
+
+
+class FastSpeech2ConformerDimensionalLayerNorm(nn.Module):
+    def __init__(self, output_dim, dim=1):
+        super().__init__()
+        self.layer_norm = nn.LayerNorm(output_dim)
+        self.dim = dim
+
+    def forward(self, predictor_activations):
+        normalized_predictor_activations = predictor_activations.transpose(self.dim, -1)
+        normalized_predictor_activations = self.layer_norm(normalized_predictor_activations)
+        normalized_predictor_activations = normalized_predictor_activations.transpose(self.dim, -1)
+        return normalized_predictor_activations
+
+
+class FastSpeech2ConformerVariancePredictorLayer(nn.Module):
+    def __init__(self, input_channels, num_chans, kernel_size, dropout_rate):
+        super().__init__()
+        self.layer = nn.ModuleList(
+            [
+                nn.Conv1d(
+                    input_channels,
+                    num_chans,
+                    kernel_size,
+                    stride=1,
+                    padding=(kernel_size - 1) // 2,
+                    bias=True,
+                ),
+                nn.ReLU(),
+                FastSpeech2ConformerDimensionalLayerNorm(num_chans),
+                nn.Dropout(dropout_rate),
+            ]
+        )
+
```

Comment:
is there a reason you are not simply using `self.conv1, self.act_fn, self.layer_norm` and remove the need for `FastSpeech2ConformerDimensionalLayerNorm`, and improve clarity?